internal/server/statvar/statvar_group.go

```diff
@@ -116,20 +115,29 @@ func filterSVG(in *pb.StatVarGroups, statVars []string) *pb.StatVarGroups {
 	return result
 }
 
-// adjustDescendentStatVarCount returns the DescendentStatVarCount for node curID.
-// Mutates DescendentStatVarCount for all nodes in id2Node starting at customSvgRoot.
-func adjustDescendentStatVarCount(id2Node map[string]*pb.StatVarGroupNode, curID string) int32 {
+// adjustDescendentStatVarCount returns all the unique stat var under curID.
+// Recursively mutates DescendentStatVarCount for all nodes in id2Node.
+func adjustDescendentSVCount(
+	id2Node map[string]*pb.StatVarGroupNode, curID string,
+) map[string]struct{} {
 	curNode, ok := id2Node[curID]
 	if !ok {
-		return 0
+		return nil
+	}
+	descendentSV := map[string]struct{}{}
+	for _, sv := range curNode.ChildStatVars {
+		descendentSV[sv.Id] = struct{}{}
 	}
-	curNode.DescendentStatVarCount = int32(len(curNode.ChildStatVars))
 	for _, childSVG := range curNode.GetChildStatVarGroups() {
+		childDescendentSV := adjustDescendentSVCount(id2Node, childSVG.GetId())
 		// Child SVG protos have its own DescendentStatVarCount that must be set.
-		childSVG.DescendentStatVarCount = adjustDescendentStatVarCount(id2Node, childSVG.GetId())
-		curNode.DescendentStatVarCount += childSVG.DescendentStatVarCount
+		childSVG.DescendentStatVarCount = int32(len(childDescendentSV))
```

Comment:
Slightly puzzled why the setting in line 139 doesn't handle this?  I'm assuming `childSVG` is same as `id2Node[childSVG.GetId()]` ?

Reply:
The structure is like:
```
{
  DescendentStatVarCount: 20,
  GetChildStatVarGroups: [
  {
     "Id": "1"
     "DescendentStatVarCount": 2
   },
  {
     "Id": "2"
     "DescendentStatVarCount": 3
   },
  ]
}
```

So there are two DescendentStatVarCount to set, this one being the children count.

Reply:
Ahh ok.  I think I was looking at the cache proto, but the mixer proto has additional field :)